### ISSUE TITLE
scopelint missing features in exportloopref

### ISSUE THREAD

**Issue by invidian** [golangci/golangci-lint#2088]
### Welcome

- [X] Yes, I'm using a binary release within 2 latest major releases. Only such installations are supported.
- [X] Yes, I've searched similar issues on GitHub and didn't find any.
- [X] Yes, I've included all information below (version, config, etc).
- [X] Yes, I've tried with the standalone linter if available. (https://golangci-lint.run/usage/linters/)

### Description of the problem

Code below triggers `scopelint` linter, which is currently deprecated. Deprecation message says that is has been replaced by `exportloopref`, but this doesn't seem 100% correct, as this linter does not trigger on the code below.

I believe this specific example below is important for the test code to be correct.

### Version of golangci-lint

<details>

```console
$ golangci-lint --version
golangci-lint has version 1.41.1 built from a2074809 on 2021-06-19T16:01:50Z
```

</details>

### Configuration file

<details>

```console
$ cat .golangci.yml
output:
  sort-results: true

run:
  timeout: 10m
  build-tags:
    - e2e
    - integration

issues:
  exclude-use-default: false
  max-same-issues: 0
  max-issues-per-linter: 0
  exclude-rules:
    # False positive: https://github.com/kunwardeep/paralleltest/issues/8.
    - linters:
      - paralleltest
      text: "does not use range value in test Run"
    - linters:
      - godot
      source: "(front proxy CA certificate,|certificate, as recommended by)"
    # Allow packages to print output to the user until we introduce loggers.
    - linters:
      - forbidigo
      text: "use of `fmt.Printf` forbidden by pattern"
    - linters:
      - forbidigo
      text: "use of `fmt.Println` forbidden by pattern"
    # Allow replacements to force specific versions of dependencies.
    - linters:
      - gomoddirectives
      text: "replacement are not allowed"
    # Allow some errors to be not wrapped, as it is not ambiguous where they come from.
    - linters:
      - wrapcheck
      text: "error returned from external package is unwrapped"
    # Allow wrapping function which return errors.
    - linters:
      - wrapcheck
      text: "error returned from interface method should be wrapped"
    # Allow certain abbreviations in JSON fields for better readability.
    - linters:
      - tagliatelle
      text: "(CIDR|CA|DNS|API|CN|SSH).+(Cidr|Ca|Dns|Api|Cn|Ssh)"

linters:
  disable:
    # Triggers on TODOs in the code, which is fine to put.
    - godox
    # As some lines are long because of the type names, and breaking it down decreases redability.
    - lll
    # Disabled until tests are split and moved to the right file names.
    - testpackage
    # Disabled until we implement some error types and migrate to use them.
    - goerr113
    # As we use formatting rules from different linter and they are conflicting.
    - gci
    # To be able to make use of Go zero-value feature.
    - exhaustivestruct
    # This linter has been deprecated.
    #- interfacer
    # This linter has been deprecated.
    #- maligned
    # This linter has been deprecated.
    #- scopelint
    # This linter has been deprecated.
    #- golint
  enable:
    - interfacer
    - maligned
    - scopelint
    - golint
    - asciicheck
    - bodyclose
    - cyclop
    - deadcode
    - depguard
    - dogsled
    - dupl
    - durationcheck
    - errcheck
    - errorlint
    - exhaustive
    - exportloopref
    - forbidigo
    - forcetypeassert
    - funlen
    - gochecknoglobals
    - gochecknoinits
    - gocognit
    - goconst
    - gocritic
    - gocyclo
    - godot
    - gofmt
    - gofumpt
    - goheader
    - goimports
    - gomnd
    - gomoddirectives
    - gomodguard
    - goprintffuncname
    - ifshort
    - importas
    - ineffassign
    - makezero
    - misspell
    - nakedret
    - nestif
    - nilerr
    - nlreturn
    - noctx
    - nolintlint
    - paralleltest
    - prealloc
    - predeclared
    - promlinter
    - revive
    - rowserrcheck
    - sqlclosecheck
    - structcheck
    - stylecheck
    - tagliatelle
    - thelper
    - tparallel
    - typecheck
    - unconvert
    - unparam
    - varcheck
    - wastedassign
    - whitespace
    - wrapcheck
    - wsl
```

</details>

### Go environment

<details>

```console
$ go version && go env
go version go1.16.5 linux/amd64
GO111MODULE=""
GOARCH="amd64"
GOBIN="/home/invidian/.local/bin"
GOCACHE="/home/invidian/.cache/go-build"
GOENV="/home/invidian/.config/go/env"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOINSECURE=""
GOMODCACHE="/home/invidian/.cache/go/pkg/mod"
GONOPROXY=""
GONOSUMDB=""
GOOS="linux"
GOPATH="/home/invidian/.cache/go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/usr/lib/go"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/lib/go/pkg/tool/linux_amd64"
GOVCS=""
GOVERSION="go1.16.5"
GCCGO="gccgo"
AR="ar"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD="/home/invidian/repos/libflexkube/go.mod"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build867212807=/tmp/go-build -gno-record-gcc-switches"
```

</details>

### Verbose output of running

<details>

```console
$ golangci-lint cache clean
$ golangci-lint run -v
# paste output here
```

</details>

### Code example or link to a public repository

<details>

```go
package main

import (
        "fmt"
        "testing"
)

func TestFunctionRangeNotUsingRangeValueInTDotRun(t *testing.T) {
        t.Parallel()

        testCases := map[string]struct {
                value string
        }{
                "1": {value: "foo"},
                "2": {value: "bar"},
        }

        for n, tc := range testCases {
                t.Run(n, func(t *testing.T) {
                        t.Parallel()
                        fmt.Println(tc.value)
                })
        }
}
```

</details>

**Comment by ldez** [comment]
Hello,

it's not really an issue of golangci-lint: we are not responsible for the deprecation of scopelint, we just put the nearest linter as replacement.
